tests/kernel/mlx5_sriov.pm

```diff
@@ -0,0 +1,110 @@
+#          For now, this only supports Mellanox ConnectX-5 and assumes
+#          this hardware is present!
+#
+# Maintainer: Michael Moese <mmoese@suse.de>
+
+use base 'opensusebasetest';
+use strict;
+use warnings;
+use testapi;
+use utils;
+
+
+sub setup_sriov {
+    my $self        = shift;
+
+    # specify this as ip/subnet, i.e. 192.168.20.1/24
+    my $ipoib_hostip = get_var('MLX5_IPOIB_HOSTIP');
+
+    # default to 2 virtual functions per physical function
+    # set it to a higher if you want to use more VM's
+    my $num_vfns = get_var('MLX5_VIRT_FN', 2); 
+
+    assert_script_run("echo $num_vfns > /sys/class/infiniband/mlx5_0/device/sriov_numvfs");
+    assert_script_run("echo $num_vfns > /sys/class/infiniband/mlx5_1/device/sriov_numvfs");
+
+    my @vfns = split(' ', script_output('lspci -D | grep Mellano | grep Virtual | awk \'{ print $1}\''));
+
+    # configure basic IPoIB
+    script_run("ip addr del $ipoib_hostip dev ib0 || true");
+    assert_script_run("ip addr add $ipoib_hostip dev ib0");
+
+    # make sure we have the driver bound to all virtual functions
+    foreach (@vfns) {
+        script_run("echo $_ > /sys/bus/pci/drivers/mlx5_core/bind 2>/dev/null || true");
+    }
+
+    my $base_guid = hex(script_output('ibstat mlx5_0; ibstat mlx5_1 ) | grep "Node GUID:" | awk \'{ print $NF}\' | sort  | tail -1 | sed -e s/0x//'));
+    my $guid = $base_guid + 1;
+
+    for $interface in ('ib0', 'ib1') {
+        for (i=0; i<$num_vfns; i++) {
+            my $clean_guid = $guid;
+            $clean_guid =~ s/([0-9A-F][0-9A-F])/\1:/g; s/:$//
+            $guid = $guid + 1;
+            assert_script_run("ip link set $interface vf $i state auto");
+            assert_script_run("ip link set $interface vf $i port_guid $clean_guid");
+            assert_script_run("ip link set $interface vf $i node_guid $clean_guid");
+        }
+        assert_script_run("ip link set $interface up");
+    }
+
+    # finally, rebind the driver for each function in order to apply the
+    # changes
+    foreach (@vfns) {
+        assert_script_run("echo $_ > /sys/bus/pci/drivers/mlx5_core/unbind");
+        assert_script_run("echo $_ > /sys/bus/pci/drivers/mlx5_core/bind");
+    }
+}
+
+sub start_vms {
+    my $self = shift;
+
+    my $ver = get_required_var('VERSION');
+    my $arch = get_required_var('ARCH');
+    my $build = get_required_var('BUILD');
+
+    my @image = ('SLES-', $ver, '-',$arch, '-', $build, '@64bit-smp-minimal_with_sdk', $build, '_installed.qcow2');
+    my $image = join '', @image;
+    assert_script_run('mkdir /mnt/openqa');
+    assert_script_run('mount -t nfs openqa.suse.de:/var/lib/openqa/share/factory /mnt/openqa');
+
+    assert_script_run('mkdir -p /var/lib/libvirt/images');
+    assert_script_run("cp $image /var/lib/libvirt/images/");
+
+    # for each desired image do:
+    #   qemu-img create -f qcow2 \
+    #            -o backing_file=/var/lib/libvirt/$image \
+    #            /var/lib/libvirt/$imageNN.qcow2
+
```

Comment:
qcow images generated by openQA usually wait at grub for keyboard input, do you count with that?

Reply:
That's actually a good point. No, I did not have this in mind - maybe I need to see if I need an alternative here.